### PR TITLE
list unavailable backends as JSON in fed errors

### PR DESCRIPTION
**Pull request by stefanwire**
## Checklist

 - [x] Add a new entry in an appropriate subdirectory of `changelog.d`
 - [x] Read and follow the [PR guidelines](https://docs.wire.com/developer/developer/pr-guidelines.html)

### CODE REVIEW

**Review by battermann — APPROVED** [review]
LGTM

File: services/brig/docs/swagger.md [hunk 1]
```diff
@@ -57,6 +57,7 @@ For errors that are more likely to be transient, we suggest clients to retry wha
 Errors in this category result from trying to communicate with a backend that is considered non-existent or invalid. They can result from invalid user input or client issues, but they can also be a symptom of misconfiguration in one or multiple backends. These errors have a 4xx status code.
 
  - **Remote backend not found** (status: 422, label: `invalid-domain`): This backend attempted to contact a backend which does not exist or is not properly configured. For the most part, clients can consider this error equivalent to a domain not existing, although it should be noted that certain mistakes in the DNS configuration on a remote backend can lead to the backend not being recognized, and hence to this error. It is therefore not advisable to take any destructive action upon encountering this error, such as deleting remote users from conversations.
+ - **Remote backend unreachable** (status: 403, label: `federation-unreachable-domains-error`): Similar to `invalid-domain`.
```
Comment (mdimjasevic) [inline services/brig/docs/swagger.md:60]:
```suggestion
 - **Remote backend unreachable** (status: 503, label: `federation-unreachable-domains-error`): Similar to `invalid-domain`. However, unlike with `invalid-domain`, here the configuration for one or more remote backends is valid, but they are temporarily unreachable.
```

**Review by mdimjasevic — COMMENTED** [review]
It's not clear from this PR how extra fields (the "data") for other federation errors is affected, e.g., if they can be considered backwards-compatible. My first hunch is that this breaks clients, but I could be wrong.

File: services/brig/docs/swagger.md (hunk 1 above)
Comment (stefanwire) [reply services/brig/docs/swagger.md]: Nice catch, thanks!

File: libs/wire-api-federation/src/Wire/API/Federation/Error.hs [hunk 2]
```diff
@@ -322,11 +323,12 @@ federationUnexpectedError msg =
...
 federationUnreachableError :: Set Domain -> Wai.Error
-federationUnreachableError (Set.map domainText -> ds) =
-  Wai.mkError
+federationUnreachableError (Set.toList -> ds) =
+  Wai.Error
     status
     "federation-unreachable-domains-error"
-    ("The following domains are unreachable: " <> (LT.pack . show . Set.toList) ds)
+    ("The following domains are unreachable: " <> (LT.pack . show . map domainText) ds)
+    (flip FederationErrorData T.empty <$> NE.nonEmpty ds)
```
Comment (mdimjasevic) [inline libs/wire-api-federation/src/Wire/API/Federation/Error.hs:331]: I haven't thought about it much, but I suppose it's OK to leave out the info on the path of the failing request in this case.

**Review by mdimjasevic — APPROVED** [review]

**Pull request merged**